tooling/templatize/pkg/pipeline/arm.go

```diff
@@ -15,23 +16,76 @@ import (
 )
 
 type armClient struct {
-	creds          *azidentity.DefaultAzureCredential
-	SubscriptionID string
-	Region         string
+	deploymentClient        *armresources.DeploymentsClient
+	resourceGroupClient     *armresources.ResourceGroupsClient
+	deploymentRetryWaitTime int
+
+	Region        string
+	GetDeployment func(ctx context.Context, rgName, deploymentName string) (armresources.DeploymentsClientGetResponse, error)
 }
 
 func newArmClient(subscriptionID, region string) *armClient {
 	cred, err := azidentity.NewDefaultAzureCredential(nil)
 	if err != nil {
 		return nil
 	}
+	deploymentClient, err := armresources.NewDeploymentsClient(subscriptionID, cred, nil)
+	if err != nil {
+		return nil
+	}
+	resourceGroupClient, err := armresources.NewResourceGroupsClient(subscriptionID, cred, nil)
+	if err != nil {
+		return nil
+	}
 	return &armClient{
-		creds:          cred,
-		SubscriptionID: subscriptionID,
-		Region:         region,
+		deploymentClient:        deploymentClient,
+		deploymentRetryWaitTime: 15,
+		resourceGroupClient:     resourceGroupClient,
+		Region:                  region,
+		GetDeployment: func(ctx context.Context, rgName, deploymentName string) (armresources.DeploymentsClientGetResponse, error) {
+			return deploymentClient.Get(ctx, rgName, deploymentName, nil)
+		},
 	}
 }
 
+func (a *armClient) getExistingDeployment(ctx context.Context, rgName, deploymentName string) (*armresources.DeploymentsClientGetResponse, error) {
+	resp, err := a.GetDeployment(ctx, rgName, deploymentName)
+	if err != nil && !strings.Contains(err.Error(), "ERROR CODE: DeploymentNotFound") {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+func (a *armClient) waitForExistingDeployment(ctx context.Context, timeOutInSeconds int, rgName, deploymentName string) error {
+	var currentTimeout = timeOutInSeconds
+	canContinue := false
+
+	resp, err := a.getExistingDeployment(ctx, rgName, deploymentName)
+	if err != nil {
+		return fmt.Errorf("Error getting deployment %w", err)
+	}
+	if resp.Properties == nil {
+		return nil
+	}
+
+	for !canContinue {
+		if *resp.Properties.ProvisioningState == armresources.ProvisioningStateRunning {
+			time.Sleep(time.Duration(a.deploymentRetryWaitTime) * time.Second)
+			currentTimeout -= a.deploymentRetryWaitTime
+			if currentTimeout < 0 {
+				return fmt.Errorf("Timeout exeeded waiting for deployment %s in rg %s", deploymentName, rgName)
+			}
+		} else {
+			return nil
+		}
+		resp, err = a.getExistingDeployment(ctx, rgName, deploymentName)
```

Comment:
Since you're already checking for the existence of the deployment at the top of the method, is this necessary here?

Reply:
I refactored the method